src/libstore/globals.hh

```diff
@@ -313,6 +313,14 @@ public:
     Setting<Strings> trustedUsers{this, {"root"}, "trusted-users",
         "Which users or groups are trusted to ask the daemon to do unsafe things."};
 
+    Setting<unsigned int> ttlNegativeDiskCache{this, 3600, "negative-disk-cache-ttl",
```

Comment:
One more thing: "disk cache" is a bit ambiguous as there can be many disk cache (e.g. `nix search` also has a disk cache). Maybe it should be called `narinfo cache` (e.g. `narinfo-cache-negative-ttl`).

Reply:
sounds good